boards/arc/hsdk4xd/doc/index.rst

```diff
@@ -0,0 +1,524 @@
+
+In the default configuration, Zephyr's HS4x/HS4xD Development Kit images support
+serial output via the USB-UART on the board.  To enable serial output:
+
+* Open a serial port emulator (i.e. on Linux minicom, putty, screen, etc)
+
+* Specify the tty driver name, for example, on Linux this may be
+  :file:`/dev/ttyUSB0`
+
+* Set the communication settings to:
+
+
+========= =====
+Parameter Value
+========= =====
+Baud:     115200
+Data:     8 bits
+Parity:    None
+Stopbits:  1
+========= =====
+
+Debugging
+==========
+
+Using the latest version of Zephyr SDK(>=0.15.2), you can debug and
+flash (run) HS4x/HS4xD Development Kit directly.
+
+One option is to build and debug the application using the usual
+Zephyr build system commands.
+
+.. zephyr-app-commands::
+   :board: hsdk4xd
+   :app: <my app>
+   :goals: debug
+
+At this point you can do your normal debug session. Set breakpoints and then
+:kbd:`c` to continue into the program.
+
+The other option is to launch a debug server, as follows.
+
+.. zephyr-app-commands::
+   :board: hsdk4xd
+   :app: <my app>
+   :goals: debugserver
+
+Then connect to the debug server at the HS4x/HS4xD Development Kit from a second
+console, from the build directory containing the output :file:`zephyr.elf`.
+
+.. code-block:: console
+
+   $ cd <my app>
+   $ $ZEPHYR_SDK_INSTALL_DIR/arc-zephyr-elf/arc-zephyr-elf-gdb zephyr.elf
+   (gdb) target remote localhost:3333
+   (gdb) load
+   (gdb) b main
+   (gdb) c
+
+Flashing
+========
+
+If you just want to download the application to the HS4x/HS4xD Development Kit's DDR
+and run, you can do so in the usual way.
+
+.. zephyr-app-commands::
+   :board: hsdk4xd
+   :app: <my app>
+   :goals: flash
+
+This command still uses openocd and gdb to load the application elf file to
+HS4x/HS4xD Development Kit, but it will load the application and immediately run. If
+power is removed, the application will be lost since it wasn't written to flash.
+
+Most of the time you will not be flashing your program but will instead debug
+it using openocd and gdb. The program can be download via the USB cable into
+the code and data memories.
+
+The HS4x/HS4xD Development Kit also supports flashing the Zephyr application
+with the U-Boot bootloader, a powerful and flexible tool for loading
+an executable from different sources and running it on the target platform.
+
+The U-Boot implementation for the HS4x/HS4xD Development Kit was further extended with
+additional functionality that allows users to better manage the broad
+configurability of the HS4x/HS4xD Development Kit
+
+When you are ready to deploy the program so that it boots up automatically on
+reset or power-up, you can follow the steps to place the program on SD card.
+
+For details, see: `Uboot-HS4x/HS4xD-Command-Reference
+<https://github.com/foss-for-synopsys-dwc-arc-processors/linux/wiki/Uboot-HSDK-4xD-Command-Reference#launching-baremetal-application-on-hsdk-4xd>`__
+
+
+Release Notes
+*************
+
```

Comment:
Empty section?
I suggest to add a table here indicating what HS4xD Development Kit features/peripherals are supported by the Zephyr BSP.